lib/dialyzer/src/dialyzer_coordinator.erl

```diff
@@ -135,22 +136,20 @@ parallel_job(Mode, Jobs, InitData, Timing) ->
 
 %% Helper for 'sigtype' and 'dataflow' workers.
 wait_for_success_typings(Labels, {_Collector, _Regulator, JobLabelsToPid}) ->
-  F =
-    fun(JobLabel) ->
-      %% The jobs that job depends on have always been started.
-      try ets:lookup_element(JobLabelsToPid, JobLabel, 2) of
-        Pid when is_pid(Pid) ->
-          Ref = erlang:monitor(process, Pid),
-          receive
-            {'DOWN', Ref, process, Pid, _Info} ->
+  F = fun(JobLabel) ->
+          %% The jobs that job depends on have always been started.
+          case ets:lookup_element(JobLabelsToPid, JobLabel, 2, ok) of
+            Pid when is_pid(Pid) ->
+              Ref = erlang:monitor(process, Pid),
+              receive
+                {'DOWN', Ref, process, Pid, _Info} ->
+                  ok
+              end;
+            ok ->
+              %% Already finished.
               ok
           end
-      catch
-        _:_ ->
-          %% Already finished.
-          ok
-      end
-    end,
+      end,
   lists:foreach(F, Labels).
```

Comment:
Why not get rid of the fun while we're at it? It seems a bit redundant.
```erlang
wait_for_success_typings(Labels, {_Collector, _Regulator, JobLabelsToPid}) ->
    wait_for_success_typings_1(Labels, JobLabelsToPid).

wait_for_success_typings_1([JobLabel | Labels], JobLabelsToPid) ->
   %% The jobs that job depends on have always been started.
    case ets:lookup_element(JobLabelsToPid, JobLabel, 2, ok) of
        Pid when is_pid(Pid) ->
            Ref = erlang:monitor(process, Pid),
            receive
                {'DOWN', Ref, process, Pid, _Info} -> ok
            end;
        ok ->
            %% Already finished.
            ok
    end,
    wait_for_success_typings_1(Labels, JobLabelsToPid);
wait_for_success_typings_1([], _JobLabelsToPid) ->
    ok.
```

Reply:
I don't see any particular problem with the fun. The host function is not very large and the fun is not too deeply indented (which are things that annoy me).

If I had written the function myself from scratch, I would probably have written something like this (using a list comprehension only for its side effects, which I know that not everybody approves of):

```
wait_for_success_typings(Labels, {_Collector, _Regulator, JobLabelsToPid}) ->
    _ = [wait_for_success_typing(Label, JobLabelsToPid) || Label <- Labels],
    ok.
```